core/src/communication/application/schema.py

```diff
@@ -0,0 +1,68 @@
+from typing import Self
+
+from pydantic import BaseModel, DirectoryPath, FilePath, field_validator, model_validator
+
+from core.src.utils.types import FiniteFloat, PositiveFiniteFloat, PositiveInteger
+
+
+class GodotSettings(BaseModel):
+    class Config:
+        allow_mutation = False
+
+    # Path leading to godot executable
+    godot_executable: FilePath
+
+    # Path leading to godot project root
+    project_path: DirectoryPath
+
+    # noinspection PyNestedDecorators
+    @field_validator("godot_executable", mode="after")
+    @classmethod
+    def validate_godot_executable(cls, value: FilePath) -> FilePath:
+        if value.suffix == ".exe":
+            return value
+        raise ValueError(f"Path should point to an .exe file but instead pointed to {value.suffix}")
+
+
+class TrainingSettings(BaseModel):
+    class Config:
+        allow_mutation = False
+
+    class Resources(BaseModel):
+        class Config:
+            allow_mutation = False
+
+        number_of_workers: PositiveInteger
+        number_of_env_per_worker: PositiveInteger
+        training_checkpoint_frequency: PositiveInteger
+
+    class TrainingVariables(BaseModel):
+        class Config:
+            allow_mutation = False
+
+        training_iterations: PositiveInteger
+        training_batch_size: PositiveInteger
+        learning_rate: PositiveFiniteFloat
+        entropy_coefficient: PositiveFiniteFloat
```

Comment:
I tu chcemy tylko to lr i entorpy_coeff, przekazywane rozumiem z Godota a reszta bedzie ustawiona, żeby pasowała do naszego środowiska? Jeszcze chyba potrzebujemy wyboru modelu do trenowania?

Reply:
I też to Custom model zrobiłam, to je łatwo przerobić że można wybrać ilość i wielkości warstw to by też można było wybierać w godocie i tez dropout i inne tego typu rzeczy.